src/libpmemfile-posix/HACKING.txt

```diff
@@ -0,0 +1,39 @@
+Inode/vinode accessing rules:
+- When accessing vinode or inode you must hold a reference on vinode.
+- Unless there's a good reason to not do this, refing should be done outside of
+  pmemobj transaction.
+- It's safe to ref in the transaction if there's nothing after ref that can
+  abort.
+  If transaction can fail after ref, you need to handle that and unref it.
+  Also in this case you have to deal with inode_ref sometimes taking a reference
+  on its parent (if it's a directory).
+  This is very fragile and tricky, so refing inside of transaction is highly
+  discouraged.
```

Comment:
FYI, the next PR will remove refing from inside of transactions, making this long point obsolete.